### DISCUSSION TITLE
ERROR - arn:aws:iam::724*****:user/Abhishek11**** is not authorized to perform: sts:AssumeRole on resource.

### DISCUSSION THREAD

**Discussion by Abseek** [amzn/selling-partner-api-models#3620] [General] [answered]
![Screenshot (58)](https://github.com/amzn/selling-partner-api-docs/assets/87055346/563c89cb-14c1-4319-b930-9dd0768dc86f)

I am getting this error while using SP-API with Python. I have added sts policy to my AWS arn still its showing not authorized to perform.

Code - 
`
from sp_api.base import Marketplaces
from sp_api.api import Orders,Tokens
from sp_api.api import Catalog

credentials = dict(
    refresh_token='Atzr|**********************************************',  # From Seller central under Authorise -> Refresh Token
    lwa_app_id='amzn1.application-oa2-client.******************',  # From Seller Central, named CLIENT IDENTIFIER on website.
    lwa_client_secret='amzn1.oa2-cs.**************************************',  # From Seller Central, named CLIENT SECRET on website.
    aws_access_key='*******************',  # From AWS IAM Setup
    aws_secret_key='****************',  # From AWS IAM Setup
    role_arn='arn:aws:iam::************'  #arn:aws:iam::1234567890:role/SellingPartnerAPIRole
)

order_client = Catalog(credentials=credentials, marketplace=Marketplaces.US)
print(Catalog(credentials=credentials).list_categories())

`
SP API Documentation Python - https://sp-api-docs.saleweaver.com/quickstart/

**Comment by rugved1991** [comment, marked as answer]
Hi @Abseek,

Please refer to this video to make sure you have followed the correct IAM configuration steps:
https://www.youtube.com/watch?v=_fppjXFEO7A&t=26s

Best,
Rugved
Solutions Architect, SP API

**Comment by Abseek** [comment]
Thanks Rugved, Got the solution.
